python/paddle/tensor/linalg.py

```diff
@@ -3724,3 +3724,133 @@ def cdist(
     return paddle.linalg.norm(
         x[..., None, :] - y[..., None, :, :], p=p, axis=-1
     )
+
+
+def householder_product(A, tau, name=None):
```

Comment:
according to [API naming conventions](https://www.paddlepaddle.org.cn/documentation/docs/zh/dev_guides/api_contributing_guides/api_design_guidelines_standard_cn.html), enter the name of Tensor using `x`, and the [rfc](https://github.com/PaddlePaddle/community/blob/master/rfcs/APIs/20230928_api_design_for_householder_product.md) should also be modified synchronously